src/main/java/com/hortonworks/spark/sql/hive/llap/HWConf.java

```diff
@@ -41,11 +41,18 @@ static String warehouseKey(String suffix) {
   }
 
   String getString(HiveWarehouseSessionState state) {
-    return Optional.
-      ofNullable((String) state.props.get(qualifiedKey)).
-      orElse(state.session.sparkContext().getConf().get(
-        qualifiedKey, (String) defaultValue)
-      );
+      String value = (String) state.props.get(qualifiedKey);
+      if (value == null) {
+          // There seems no way to call spark.conf.get(key, default)?
+          if (state.session.conf().getOption(qualifiedKey).nonEmpty()) {
```

Comment:
Seems weird that we can't use `spark.conf.get(key, default)` in Java side. At least I wasn't able to use it although I see no problem to use it .. I worked around by getOption.